Detectors/MUON/MCH/PreClustering/src/DigitSamplerSpec.cxx

```diff
@@ -86,19 +91,44 @@ class DigitSamplerTask
     memcpy(bufferPtr, &nDigits, SSizeOfInt);
     bufferPtr += SSizeOfInt;
 
-    // fill digits if any
+    // fill digits in O2 format, if any
     if (size > 0) {
       mInputFile.read(bufferPtr, size);
+      if (mUseRun2DigitUID) {
+        convertDigitUID2PadID(reinterpret_cast<Digit*>(bufferPtr), nDigits);
+      }
     } else {
       LOG(INFO) << "event is empty";
     }
   }
 
  private:
+  //_________________________________________________________________________________________________
+  void convertDigitUID2PadID(Digit* digits, int nDigits)
+  {
+    /// convert the digit UID in run2 format into a pad ID (i.e. index) in O2 mapping
+
+    for (int iDigit = 0; iDigit < nDigits; ++iDigit) {
+
+      int deID = digits[iDigit].getDetID();
+      int digitID = digits[iDigit].getPadID();
+      int manuID = (digitID & 0xFFF000) >> 12;
+      int manuCh = (digitID & 0x3F000000) >> 24;
+
+      int padID = mapping::segmentation(deID).findPadByFEE(manuID, manuCh);
+      if (padID < 0) {
+        throw runtime_error(std::string("digitID ") + digitID + " does not exist in the mapping");
+      }
+
+      digits[iDigit].setPadID(padID);
+    }
+  }
+
   static constexpr uint32_t SSizeOfInt = sizeof(int);
```

Comment:
just out of curiosity : I see here and there that you are using those constants instead of just using the sizeof(xxx) in the code. I believe there's no performance difference (I guess the compiler's perfectly able to elude the sizeof) ? And for code clarity I personally find it more difficult to follow as I have to find out what SSizeOfInt really is (yes, given to -good- name I can suppose it's sizeof(int)) but I have to find back the definition to be sure...

Reply:
I don't know, I thought this way I make sure this is not computed runtime and I kept the habit of doing it. But this is nano optimization and maybe the compiler does it anyway. I can get rid of it in the next commit if you want.

Reply:
Well, AFAIK sizeof is always evaluated at compile time. Note also that it returns a std::size_t, which is most probably not uint32_t ...

Reply:
ok, I will remove them in the next PR